includes/options/views/class-amp-setup-wizard-submenu-page.php

```diff
@@ -129,18 +129,26 @@ public function enqueue_assets( $hook_suffix ) {
 
 		wp_styles()->add_data( self::ASSET_HANDLE, 'rtl', 'replace' );
 
+		$setup_wizard_data = [
+			'AMP_OPTIONS_KEY'             => AMP_Options_Manager::OPTION_NAME,
+			'APP_ROOT_ID'                 => self::APP_ROOT_ID,
+			'EXIT_LINK'                   => admin_url( 'admin.php?page=' . AMP_Options_Manager::OPTION_NAME ),
+			'OPTIONS_REST_ENDPOINT'       => rest_url( 'amp/v1/options' ),
+			'READER_THEMES_REST_ENDPOINT' => rest_url( 'amp/v1/reader-themes' ),
+			'UPDATES_NONCE'               => wp_create_nonce( 'updates' ),
+		];
+
 		wp_add_inline_script(
 			self::ASSET_HANDLE,
 			sprintf(
 				'var ampSetup = %s;',
 				wp_json_encode(
-					[
-						'APP_ROOT_ID'            => self::APP_ROOT_ID,
-						'EXIT_LINK'              => admin_url( 'admin.php?page=' . AMP_Options_Manager::OPTION_NAME ),
-						'OPTIONS_REST_ENDPOINT'  => rest_url( 'amp/v1/options' ),
-						'READER_THEMES_ENDPOINT' => rest_url( 'amp/v1/reader-themes' ),
-						'UPDATES_NONCE'          => wp_create_nonce( 'updates' ),
-					]
+					/**
+					 * Filters the array of data passed to the setup wizard application.
+					 *
+					 * @param array $setup_wizard_data Associateive array of data.
+					 */
+					apply_filters( 'amp_setup_wizard_data', $setup_wizard_data )
```

Comment:
Introducing a global filter here doesn't seem ideal, as it would allow other plugins to inject arbitrary things here.

I believe the what @schlessera will recommend is that the `UserManager` instance (or however we call it) be supplied as a dependency for this class, and then we can ask for the additional fields via `$this->userManager->getExportedSetupData()`

Reply:
I removed this filter and ended up just putting the new localized data right into the array in this class. There was really no need to use the filter when this class is already passing different types of data into the JS for the setup wizard, and I couldn't see a good reason to inject the new class as a dependency for this one purpose that isn't useful anywhere else.